src/Cohesity.Powershell/Scripts/Utility/Invoke-RestApi.ps1

```diff
@@ -10,10 +9,41 @@ function Invoke-RestApi
         $Body,
         $OutFile
     )
-    if($null -eq $Global:CohesityCmdletConfig) {
+    if ($null -eq $Global:CohesityCmdletConfig) {
         $Global:CohesityCmdletConfig = Get-CohesityCmdletConfig
     }
-    if($null -eq $Global:CohesityUserAgentName) {
+
+    if ($Global:CohesityCmdletConfig) {
+        if ($true -eq $Global:CohesityCmdletConfig.RefreshToken) {
+            $cohesitySession = Get-Content -Path $HOME/.cohesity | ConvertFrom-Json
+            # The system saves the utc time in nano seconds
+            $timeDiff = New-TimeSpan $cohesitySession.TimestampUTC (Get-Date).ToFileTimeUtc()
+            if ( $timeDiff.TotalHours -ge 24) {
```

Comment:
this is not a long running process right ? why are we checking for the timestamp ? We should just catch the exception of expired access_token.

Reply:
The timestamp is introduced because of the limitations with powershell and the cmdlet framework. All the calls are one way which ends up in Invoke-RestApi and hence we have the above design.
This is what we had agreed, if the time lapse is more than 24 hours refresh the token.

Reply:
if time lapse is > 24 hours, the access token exception is thrown anyways. I fail to see the point of having timestamp based token change.

Reply:
Will use the error code "Unauthorized" to make a fresh attempt to refresh the session token.
Code review feedback implemented.

Reply:
Yes, have a retry till 3 or 5 times though.. wcs if iris server goes down , we dont want to be retrying infinitely.